legion/engine/physics/physx/data/physx_wrapper_container.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <physics/physx/data/physx_wrapper.hpp>
+#include <physics/components/physics_component.hpp>
+
+namespace legion::physics
+{
+    class PhysxWrapperContainer
+    {
+    public:
+
+        PhysxInternalWrapper& createPhysxWrapper(physics_component& unregisteredPhysXWrapper);
+
+        void PopAndSwapRemoveWrapper(size_type id);
+
+        std::optional<std::reference_wrapper< PhysxInternalWrapper>> findWrapperWithID(size_type id);
```

Comment:
if you're returning an optional of a ref wrapper anyways? why not return a pointer? pointers will have less overhead. if you want to return non owning pointers while avoiding raw pointers, you can always use the non deletable `pointer<T>` in `core/containers`